src/feature/flow_chart_panel/views/AppGalleryModal.tsx

```diff
@@ -22,18 +22,33 @@ import {
 const subjectKeyList = ["fundamentals", "AI", "IO", "DSP"];
 const ignoreDir = [".github", "MANIFEST"];
 
-type AppGalleryModalProps = {
+export interface GithubJSON {
+  name: string;
+  path: string;
+  sha: string;
+  size: number;
+  url: string;
+  html_url: string;
+  git_url: string;
+  download_url?: string;
+  type: string;
+  links: object[];
+}
+
+interface AppGalleryModalProps {
   isGalleryOpen: boolean;
   setIsGalleryOpen: (open: boolean) => void;
-};
+}
 
 export const AppGalleryModal = ({
   isGalleryOpen,
   setIsGalleryOpen,
 }: AppGalleryModalProps) => {
-  const [selectFields, setSelect] = useState([]);
-  const [data, setData] = useState<object[]>([]);
+  const [selectFields, setSelectFields] = useState([]);
+  const [searchData, setSearchData] = useState<GithubJSON[]>([]);
```

Comment:
I was able to do this, but it seems like the search component I was using would freeze when the file headings are too much (the limit is around 15 ish?). Is it ok if I disable the search bar at first and contain a message like "please select a category" instead? Otherwise, I would need to do some big refactors to make it work...